grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -390,8 +397,9 @@ public void onNext(DeframedMessage message) {
                 listener.onMessage(msg);
             }
         } catch (Throwable t) {
-            req.close(GrpcStatus.fromThrowable(t).asException());
-            throw t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t);
+            final Status status = GrpcStatus.fromThrowable(exceptionMappings, t);
+            req.close(status.asException());
+            close(status, new Metadata());
```

Comment:
This is a bug fix. `onNext()` of `Subscriber` should return normally. 
https://github.com/reactive-streams/reactive-streams-jvm#2.13